### PR TITLE
Optionalize all non-built-ins

### LINKED ISSUE #15: All non-builtin loggers should be optional
Microservice environments might well have dozens or hundreds copies of metalogger installed; there's no reason to add this bulk when most loggers aren't needed for most installations.

- 44K using util, loggly, or ain2
- 96K with log
- 0.6MB with winston
- 0.8MB with npmlog
- 32MB with "sumologic"

...but especially sumologic, since it imports s3-streamlogger, which imports the aws-sdk for a cool 30MB for every instance.

(In fact, that logger should probably be renamed to "s3", since there are actually-direct-to-sumologic loggers out there, which metalogger isn't actually using.)

### PR DESCRIPTION
**Pull request by randallsquared**
This PR makes all loggers that have to be `require`d optional, so that not every existing logger (`s3-streamlogger`, I'm lookin' at you) or logger that `metalogger` supports in the future has to be included in each installation.  This saves over 30MB per installation, which for microservice/container usage can be significant.

However, this is definitely a breaking change, since previously a `metalogger` user could count on their logger to be present if it didn't happen to be `ain2` or `loggly`.  So, there's that.

Fixes #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,20 +60,17 @@ using Metalogger to ensure that a switch-over to a different logging implementat
 > npm install metalogger
 ```
 
-If you are planning to use syslogging via ain2 bridge, or loggly, you should
-install them manually, since they were made into optional dependencies:
+The `util` logging bridge relies on the built-in node `util`; for any other
+logging type, you must the underlying logger manually; metalogger does not
```
Comment (inadarei) [inline README.md]: A word is missing: 

> you must "install" the underlying logger manually.

Also, semicolon ";" is unnecessary. Let's just start a new sentence there.

File: README.md [hunk 2]
```diff
@@ -42,15 +42,14 @@ logging experience.
...
-[sumologic](https://www.sumologic.com/), 
+[npmlog](https://github.com/isaacs/npmlog),
+[loggly](https://www.loggly.com/),
+[sumologic](https://www.sumologic.com/),
 [syslog/ain2](https://npmjs.org/package/ain2),
-[log](https://github.com/visionmedia/log.js), 
-[util](http://nodejs.org/api/util.html). A node.js module that 
-uses the metalogger library can choose which logging implementation to use at runtime.
+[log](https://github.com/visionmedia/log.js),
+[util](http://nodejs.org/api/util.html).
```
Comment (inadarei) [inline README.md]: This used to be a list, diff shows you removed "-" characters and it's not a list, anymore? If, true - please keep the list.

Also, let's add an item for "winston" or even "any other logging facility supported by \[winston\]\(link\)

Comment (randallsquared) [inline README.md]: I can certainly make it into a list.

File: README.md (hunk 1 above)
Comment (randallsquared) [inline README.md]: Fixed.

**Comment by randallsquared** [comment]
@inadarei changes made.

**Pull request merged**